### DISCUSSION TITLE
unknown argument 'id' when trying to connect

### DISCUSSION THREAD

**Discussion by wintercounter** [prisma/prisma#20832] [Q&A]
### Question

I'm getting the following:

```js
- error PrismaClientValidationError: 
Invalid `prisma.playlist.update()` invocation:

{
  where: {
    id: "2P0SIRabey9y2LiiPaGYLF"
  },
  data: {
    tracks: {
      connect: [
        {
          id: "28k1CXv1zkGRXzRNIYOsYt"
        },
        {
          id: "1HH16NEQKUQGqW4GzQPXWr"
        }
      ]
    }
  }.
  include: {
    tracks: true
  }
}

Unknown argument `id`. Did you mean `OR`? Available options are listed in green.
```

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
model Playlist {
  id          String              @id
  images      String
  tracks      TracksOnPlaylists[]
  href        String
  name        String
  description String
  uri         String
  meta        String
}

model Track {
  id                String              @id
  href              String
  name              String
  previewUrl        String?
  uri               String
  duration          Int
  explicit          Boolean
  artists           Artist[]
  meta              String
  popularity        Int
  addedAt           String
  releasedAt        String
  TracksOnPlaylists TracksOnPlaylists[]
  TracksOnArtists   TracksOnArtists[]
}

model TracksOnPlaylists {
  playlistId String
  trackId    String
  Playlist   Playlist @relation(fields: [playlistId], references: [id])
  Track      Track    @relation(fields: [trackId], references: [id])

  @@id([trackId, playlistId])
}
```

```ts
await prisma.playlist.update({
    where: {
        id: playlist.id
    },
    data: {
        tracks: {
            connect: playlist.trackIds.map(id => ({ id }))
        }
    },
    include: {
        tracks: true
    }
})
```

- OS: WSL2
- Database: SQLite
- Node.js version: 20

prisma                  : 5.2.0
@prisma/client          : 5.2.0
Current platform        : debian-openssl-3.0.x
Query Engine (Node-API) : libquery-engine 2804dc98259d2ea960602aca6b8e7fdc03c1758f (at node_modules/@prisma/engines/libquery_engine-debian-openssl-3.0.x.so.node)
Schema Engine           : schema-engine-cli 2804dc98259d2ea960602aca6b8e7fdc03c1758f (at node_modules/@prisma/engines/schema-engine-debian-openssl-3.0.x)
Schema Wasm             : @prisma/prisma-schema-wasm 5.2.0-25.2804dc98259d2ea960602aca6b8e7fdc03c1758f
Default Engines Hash    : 2804dc98259d2ea960602aca6b8e7fdc03c1758f
Studio                  : 0.494.0

**Comment by ludralph** [comment]
Hi @wintercounter 👋 

Apologies for the late reply here. Is this question still relevant or you have found a solution already?

**Reply by parmetra** [reply]
Hello @ludralph !
Please, if you don't mind, could you please write a solution to this problem, since I have a similar Database model and exactly the same problem. And I couldn't find solutions elsewhere.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
